agent/consul/server_test.go

```diff
@@ -1277,7 +1277,6 @@ func TestServer_TLSToFullVerify(t *testing.T) {
 		c.CAFile = "../../test/client_certs/rootca.crt"
 		c.CertFile = "../../test/client_certs/server.crt"
 		c.KeyFile = "../../test/client_certs/server.key"
-		c.VerifyIncoming = true
```

Comment:
This is an important change.
This test describes the scenario when during an upgrade, s2 configured for TLS (without the verify_* turned on) sends an RPC to s1 with `VerifyIncoming` enabled. It used to pass because it was using `leader.UseTLS` above for PING and leader is s1 which is the server that this test is sending the RPC to.

Since s2's `tlsConfigurator` now decides `UseTLS=false`, because the `verify_outgoing` is disabled. And since s1 has `verify_incoming=true`, they cannot speak to each other.

Which is why I changed s1 to `verify_incoming=false` to make the test pass again. Which is also the upgrade path:

1) setup CA and certs everywhere
2) setup verify_outgoing
3) setup verify_incoming